### ISSUE TITLE
RabbitMQ env sorting cause not working cluster

### ISSUE THREAD

**Issue by Nefelim4ag** [bitnami/charts#6638]
We deploy Rabbit to Rancher and sometimes we got env sorted by names (not sure why),
because ENV are resolved one by one, that caused container failed to start because of that

So, maybe the simplest solution, as your helm really **__depends__** on env ordering. we can template some part of values (like namespace)

Or just rename critical things to something like: A_MY_POD_IP to be sure, we always have correct env ordering.

Thanks!

(I can send PR for that, just not sure, that renaming or templating will be the right solution)

**Comment by miguelaeh** [comment]
Hi @Nefelim4ag ,
Could you give more information on this? Who does sort the env vars? I guess that could only cause errors if an env var depends on the value of another one. But I don't fully understand where are the env vars being sorted.

**Comment by Nefelim4ag** [comment]
@miguelaeh 
https://github.com/bitnami/charts/blob/master/bitnami/rabbitmq/templates/statefulset.yaml#L138

**Comment by miguelaeh** [comment]
Hi @Nefelim4ag ,
I just checked the envs in the chart and they seem to have the correct order. From the K8s docs:

> Environment variables may reference each other, however ordering is important. Variables making use of others defined in the same context must come later in the list. Similarly, avoid circular references.

So, sorting them alphabetically has no sense unless you resolve those issues. I think it is an issue with RAncher itself, and I found this in their repo: https://github.com/rancher/rancher/issues/16148 which is probably the same issue than in this case.
